src/hal/hal.cpp

```diff
@@ -326,11 +333,38 @@ void hal_printf_init() {
 #endif // defined(LMIC_PRINTF_TO)
 
 void hal_init (void) {
-    hal_init_ex(&lmic_pins);
+    // use the global constant
+    Arduino_LMIC::hal_init_with_pinmap(&lmic_pins);
 }
 
+// hal_init_ex is a C API routine, written in C++, and it's called
+// with a pointer to an lmic_pinmap. This is deprecated!
```

Comment:
"This is deprecated" turns out to be an obsolete comment -- I decided not to deprecate ti.